diff --git a/gyp/common.gypi b/gyp/common.gypi
index 9e53b06..3062434 100644
--- a/gyp/common.gypi
+++ b/gyp/common.gypi
@@ -95,7 +95,7 @@
         # debugging purposes.
         'configurations': {
           'Debug':   { 'defines': [ 'SK_DEVELOPER=1' ] },
-          'Release': { 'defines': [ 'NDEBUG' ] },
+          'Release': { 'defines': [ 'NDEBUG' ], 'cflags': ['-ffunction-sections', '-fdata-sections', '-fno-rtti'] },
           'Release_Developer': {
             'inherit_from': ['Release'],
             'defines': [ 'SK_DEVELOPER=1' ],
diff --git a/gyp/common_conditions.gypi b/gyp/common_conditions.gypi
index 115d94b..6f01685 100644
--- a/gyp/common_conditions.gypi
+++ b/gyp/common_conditions.gypi
@@ -254,7 +254,6 @@
           }],
           [ 'skia_warnings_as_errors', {
             'cflags': [
-              '-Werror',
             ],
           }],
           # For profiling; reveals some costs, exaggerates others (e.g. trivial setters & getters).
@@ -579,7 +578,7 @@
             'cflags': ['-g']
           },
           'Release': {
-            'cflags': ['-O2'],
+            'cflags': ['-Os', '-ffunction-sections', '-fdata-sections', '-fno-rtti'],
           },
         },
         'libraries': [
diff --git a/gyp/common_variables.gypi b/gyp/common_variables.gypi
index 52563ae..f108c3e 100644
--- a/gyp/common_variables.gypi
+++ b/gyp/common_variables.gypi
@@ -150,7 +150,7 @@
       # These variables determine the default optimization level for different
       # compilers.
       'skia_default_vs_optimization_level': 3, # full (/Ox)
-      'skia_default_gcc_optimization_level': 3, # -O3
+      'skia_default_gcc_optimization_level': 's', # -O3
     },
 
     'conditions': [
